core/src/elf_extraction.rs

```diff
@@ -0,0 +1,283 @@
+//! ELF file extraction utilities for separating ELF parsing from ZiskRom population
+
+use elf::{
+    abi::{SHF_ALLOC, SHF_EXECINSTR, SHF_WRITE, SHT_PROGBITS},
+    endian::AnyEndian,
+    ElfBytes,
+};
+use std::{error::Error, fs, path::Path};
+
+use crate::{is_elf_file, RAM_ADDR, RAM_SIZE};
+
+const RAM_START_ADDR: u64 = RAM_ADDR;
+const RAM_END_ADDR: u64 = RAM_ADDR + RAM_SIZE;
+
+/// Raw bytes of `data` that will live at `addr` once the ROM has booted.
+#[derive(Debug, Clone)]
+pub struct DataSection {
+    pub addr: u64,
+    pub data: Vec<u8>,
```

Comment:
This includes code, so maybe best to rename it to RawSection, so we don't get confused with Data as in .rodata